File: src/main/java/com/freya02/bot/docs/CachedClass.java
package com.freya02.bot.docs;

import com.freya02.docs.SeeAlso;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.List;

public final class CachedClass {
	private final MessageEmbed classEmbed;
	private final List<SeeAlso.SeeAlsoReference> seeAlsoReferences;

	public CachedClass(MessageEmbed classEmbed,
	                   List<SeeAlso.SeeAlsoReference> seeAlsoReferences) {
		this.classEmbed = classEmbed;
		this.seeAlsoReferences = seeAlsoReferences;
	}

	public MessageEmbed getClassEmbed() {
		return classEmbed;
	}

	public List<SeeAlso.SeeAlsoReference> getSeeAlsoReferences() {
		return seeAlsoReferences;
	}
}

File: doxxy-bot/src/main/kotlin/dev/freya02/doxxy/bot/docs/cached/CachedClass.kt
package dev.freya02.doxxy.bot.docs.cached

import dev.freya02.doxxy.bot.docs.index.DocIndex
import dev.freya02.doxxy.bot.docs.metadata.ImplementationIndex
import dev.freya02.doxxy.docs.sections.SeeAlso.SeeAlsoReference
import net.dv8tion.jda.api.entities.MessageEmbed

class CachedClass(
    override val docIndex: DocIndex,
    val name: String,
    override val embed: MessageEmbed,
    override val seeAlsoReferences: List<SeeAlsoReference>,
    override val sourceLink: String?,
    val subclasses: List<ImplementationIndex.Class>,
    val superclasses: List<ImplementationIndex.Class>
) : CachedDoc {
    override val qualifiedName: String
        get() = name
}
